features/dd-sdk-android-session-replay-compose/src/test/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/CheckboxSemanticsNodeMapperTest.kt

```diff
@@ -0,0 +1,517 @@
+            path = any(),
+            strokeColor = eq(fakeCheckmarkColor.toInt()),
+            strokeWidth = eq(STROKE_WIDTH_DP.toInt()),
+            targetWidth = eq(CHECKBOX_SIZE_DP),
+            targetHeight = eq(CHECKBOX_SIZE_DP),
+            density = eq(fakeDensity),
+            isContextualImage = eq(false),
+            imagePrivacy = eq(ImagePrivacy.MASK_NONE),
+            asyncJobStatusCallback = eq(mockAsyncJobStatusCallback),
+            clipping = eq(null),
+            shapeStyle = eq(expectedShapeStyle),
+            border = eq(expectedBorder),
+            customResourceIdCacheKey = eq(null)
+        )
+    }
+
+    @Test
+    fun `M return image wireframe W map { checked, reflection resolution success }`() {
+        // Given
+        whenever(mockSemanticsUtils.resolveInnerBounds(mockSemanticsNode)) doReturn fakeGlobalBounds
+
+        whenever(mockConfig.getOrNull(SemanticsProperties.ToggleableState))
+            .thenReturn(ToggleableState.On)
+
+        whenever(mockSemanticsUtils.resolveCheckboxFillColor(mockSemanticsNode))
+            .thenReturn(fakeFillColor)
+
+        whenever(mockColorUtils.parseColorSafe(fakeFillColorHexString))
+            .thenReturn(fakeFillColor.toInt())
+
+        whenever(mockColorUtils.parseColorSafe(fakeCheckmarkColorHexString))
+            .thenReturn(fakeCheckmarkColor.toInt())
+
+        whenever(
+            mockUiContext.imageWireframeHelper.createImageWireframeByPath(
+                id = any(),
+                globalBounds = eq(fakeGlobalBounds),
+                path = any(),
+                strokeColor = eq(fakeCheckmarkColor.toInt()),
+                strokeWidth = eq(STROKE_WIDTH_DP.toInt()),
+                targetWidth = eq(CHECKBOX_SIZE_DP),
+                targetHeight = eq(CHECKBOX_SIZE_DP),
+                density = eq(fakeDensity),
+                isContextualImage = eq(false),
+                imagePrivacy = eq(ImagePrivacy.MASK_NONE),
+                asyncJobStatusCallback = eq(mockAsyncJobStatusCallback),
+                clipping = eq(null),
+                shapeStyle = anyOrNull(),
+                border = anyOrNull(),
+                customResourceIdCacheKey = eq(null)
+            )
+        ).thenReturn(mock<MobileSegment.Wireframe.ImageWireframe>())
+
+        // When
+        val wireframes = testedMapper.map(
+            semanticsNode = mockSemanticsNode,
+            parentContext = mockUiContext,
+            asyncJobStatusCallback = mockAsyncJobStatusCallback
+        )
+
+        // Then
+        assertThat(wireframes.wireframes).hasSize(1)
+        assertThat(wireframes.wireframes[0]).isInstanceOf(MobileSegment.Wireframe.ImageWireframe::class.java)
+    }
+
+    @Test
+    fun `M show unchecked wireframe W map() { masked }`() {
+        // Given
+        whenever(mockUiContext.textAndInputPrivacy)
+            .thenReturn(TextAndInputPrivacy.MASK_ALL_INPUTS)
+
+        // When
+        val wireframes = testedMapper.map(
```

Comment:
since return type is `SemanticsWireframe` should this one be called `wireframe`? or `checkboxWireframe`.

Otherwise `wireframes.wireframes` below looks strange.